File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def check(num):
    result = 0
    for i in range(1, int(num**0.5) + 1):
        if i**2 == num:
            result += 1
        elif num % i == 0:
            result += 2
    return result

def solution(number, limit, power):
    answer = 0
    
    for i in range(1, number + 1):
        result = check(i)
        if result > limit:
            answer += power
        else:
            answer += result
    
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    
    const getDivisor = (num, limit, power) => {
        let result = new Set();
        for (let i = 1; i <= num**(0.5); i++) {
            if (num % i === 0) {
                result.add(i);
                result.add(num / i);
            }
            if (result.size > limit) return power;
        }
        return result.size;
    }
    
    let answer = 0;
    for (let i = 1; i <= number; i++) {
        const divisorCount = getDivisor(i, limit, power);
        answer += divisorCount;
        
    }
    return answer;
}